src/util/prolog/prologSession.ts

```diff
@@ -52,38 +71,53 @@ function getDependencyGraph(
 		return graph;
 }
 
+/** Executes the prolog programs and queries, fetches the raw results and creates the PrologResult */
 export class PrologSession {
 
-		public readonly session: any;
-
-		constructor(program: string, defaultPath: string = __dirname, fileName?: string, limit?: number) {
-				program = removeNonFunctionalities(program);
-				program = resolveImports(program, defaultPath, fileName);
-
-				this.session = new type.Session(limit);
-				this.session.consult((removeModuleDeclarations(program)));
-		}
-
-		public async executeQuery(query: string): Promise<PrologResult> {
-				this.session.query(query);
-
-				const rawResults: string[] = [];
-				let result: string;
-
-				do {
-						result = await this.getNextAnswer();
-						rawResults.push(result);
-				} while (result[result.length - 1] !== ".");
-
-				return new PrologResult(rawResults);
-		}
-
-		private getNextAnswer() {
-				return new Promise<string>((resolve) => {
-						this.session.answers((x: any) => {
-								resolve(format_answer(x));
-						});
-				});
-		}
+	public readonly session: any;
+
+	/**
+		* @param {string} program - The program code to run
+		* @param {string} defaultPath - Only needed for forwarding methode internally
+		* @param {string} fileName - Only needed for forwarding methode internally
+		* @param {number} limit - The maximum amount of resolving steps to do before aborting a query
```

Comment:
Maybe "maxIterations" would be a mote intuitive name.

Reply:
That's true. I will change it to `maxSteps` since it will be shorter and these refer to the resolving steps as in the debugger if I am not mistaken.